internal/controller/components/reconciler.go

```diff
@@ -460,3 +527,105 @@ func getFalseConditions(gp capioperatorv1.GenericProvider) []string {
 	}
 	return messages
 }
+
+func getRegistryCredsFromPullSecret(secret *corev1.Secret, registry string) (username, password string, err error) {
+	if secret.Type != "kubernetes.io/dockerconfigjson" {
+		return "", "",
+			fmt.Errorf("wrong type for imagePullSecret %q, expected kubernetes.io/dockerconfigjson", secret.Type)
+	}
+
+	data := secret.Data
+	configstr, ok := data[".dockerconfigjson"]
+	if !ok {
+		return "", "",
+			errors.New("unable to get .dockerconfigjson from imagePullSecret data")
+	}
+
+	var config dockerconfig.ConfigFile
+	if err := json.Unmarshal(configstr, &config); err != nil {
+		return "", "",
+			fmt.Errorf("failed to unmarshal dockerconfig: %w", err)
+	}
+
+	auths := config.AuthConfigs
+
+	rs := strings.Split(registry, "/")
+	registryHost := rs[0]
+
+	authConfig, ok := auths[registryHost]
+	if !ok {
+		return "", "",
+			fmt.Errorf("failed to extract auth config for the registry host %q", registryHost)
+	}
+
+	if authConfig.Auth != "" {
+		auth, err := base64.StdEncoding.DecodeString(authConfig.Auth)
+		if err != nil {
+			return "", "",
+				fmt.Errorf("unable to decode auth: %w", err)
+		}
+
+		authSplit := strings.Split(string(auth), ":")
+		if len(authSplit) != 2 {
+			return "", "",
+				errors.New("incorrect \":\" delimeted auth value")
+		}
+
+		return authSplit[0], authSplit[1], nil
+	}
+
+	if authConfig.Username != "" && authConfig.Password != "" {
+		return authConfig.Username, authConfig.Password, nil
+	}
+
+	return "", "", errors.New("unable to identify auth parameters in the dockerconfig")
+}
+
+func makeProviderSecret(username, password, componentName, namespace string, template *kcmv1.ProviderTemplate) (*corev1.Secret, bool, error) {
+	commonStatus := template.GetCommonStatus()
+
+	if commonStatus.Config == nil {
```

Comment:
I may be missing something, but template `status.config` appears to contain only the default Helm chart values, not the overrides supplied by the user. How are user-specified overrides from the Management or Region resource handled in this case?